packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1213,4 +1215,62 @@ class TextInput {
+  /// platform's text input plugin. The platform is encouraged to save the user
+  /// input stored in the current autofill context before the context is
+  /// destroyed, when [finishAutofillContext] is called with `shouldSave` set to
+  /// true.
+  ///
+  /// Currently, there can only be at most one autofill context at any given
+  /// time. When any input field in an [AutofillGroup] requests for autofill
+  /// (which is done automatically when an autofillable [EditableText] gains
+  /// focus), the current autofill context will merge the content of that
+  /// [AutofillGroup] into itself. When there isn't an existing autofill context,
+  /// one will be created to hold the newly added input fields from the group.
+  ///
+  /// Once added to an autofill context, an input field will stay in the context
+  /// until the context is destroyed. To prevent leaks, call [finishAutofillContext]
+  /// to signal the text input plugin that the user has finalized their input in
+  /// the current autofill context. The platform text input plugin either
+  /// encourages or discourages the platform from saving the user input based on
+  /// the value of the `shouldSave` parameter. The platform usually shows a
+  /// "Save for autofill?" prompt for user confirmation.
+  /// {@endtemplate}
+  ///
+  /// Typically, this method should be called when the user has finalized their
+  /// input. For example, in a [Form], it's typically done immediately before or
+  /// after its content is submitted. You should make sure that any connected
+  /// [TextInputClient] is disconnected and no input field is focused before
+  /// calling [finishAutofillContext], to ensure the user can't interact with
+  /// the input fields while it's being processed.
+  ///
+  /// The topmost [AutofillGroup]s also have the ability to call
+  /// [finishAutofillContext] automatically when they are getting disposed. The
+  /// exact behavior can be configured using [AutofillGroup.onDisposeAction].
+  ///
+  /// Calling [finishAutofillContext] may cause the platform to show the "save
+  /// for autofill" dialog and disrupts the user's flow. Ideally the dialog
+  /// should only be shown no more than once for every screen. Consider removing
+  /// premature [finishAutofillContext] calls to prevent showing the "Save for
+  /// autofill?" UI too frequently. However, calling [finishAutofillContext]
+  /// when there's no existing autofill context does not do anything.
+  ///
+  /// See also:
+  ///
+  /// * [AutofillGroup.onDisposeAction], a configurable action that runs when a
+  ///   topmost [AutofillGroup] is getting disposed.
+  static void finishAutofillContext({ bool shouldSave = true }) {
+    assert(shouldSave != null);
+    assert(TextInput._instance._currentConnection == null);
+    TextInput._instance._channel.invokeMethod<void>(
+      shouldSave
+        ? 'TextInput.AutofillContext.commit'
+        : 'TextInput.AutofillContext.cancel',
+    );
```

Comment:
Can we also change this method to send "commit/cancel" information as an argument to make it similar to the other methods?

Reply:
Update to take a bool. Let me update the engine PR.